src/simulation/in_memory_simulation_store.jl

```diff
@@ -61,110 +61,152 @@ function write_optimizer_stats!(
     store::InMemorySimulationStore,
     model_name,
     stats::OptimizerStats,
-    timestamp::Dates.DateTime,
+    index::DecisionModelIndexType,
 )
-    write_optimizer_stats!(store.dm_data[model_name], stats, timestamp)
+    write_optimizer_stats!(get_dm_data(store)[model_name], stats, index)
     return
 end
 
-# TODO EmulationModel: this interface is TBD
-#function write_optimizer_stats!(
-#    store::InMemorySimulationStore,
-#    stats::OptimizerStats,
-#    execution::Int,
-#)
-#    write_optimizer_stats!(store.em_data, stats, index)
-#    return
-#end
-
-function read_model_optimizer_stats(
+function write_optimizer_stats!(
     store::InMemorySimulationStore,
-    model_name,
-    timestamp::Dates.DateTime,
+    stats::OptimizerStats,
+    index::EmulationModelIndexType,
 )
-    return read_optimizer_stats(store.dm_data[model_name], timestamp)
+    write_optimizer_stats!(get_em_data(store), stats, index)
+    return
 end
 
-# TODO EmulationModel: this interface is TBD
-#function read_model_optimizer_stats(
-#    store::InMemorySimulationStore,
-#    execution::Int,
-#)
-#    return read_optimizer_stats(store.em_data, execution)
-#end
+function write_result!(
+    store::InMemorySimulationStore,
+    model_name::Symbol,
+    key::OptimizationContainerKey,
+    index::DecisionModelIndexType,
+    update_timestamp::Dates.DateTime,
+    array,
+)
+    write_result!(
+        get_dm_data(store)[model_name],
+        model_name,
+        key,
+        index,
+        update_timestamp,
+        array,
+    )
+    return
+end
+
+function write_result!(
+    store::InMemorySimulationStore,
+    model_name::Symbol,
+    key::OptimizationContainerKey,
+    index::EmulationModelIndexType,
+    update_timestamp::Dates.DateTime,
+    array,
+)
+    write_result!(get_em_data(store), model_name, key, index, update_timestamp, array)
+    return
+end
+
+function read_optimizer_stats(store::InMemorySimulationStore, model_name)
+    # TODO EmulationModel: this interface is TBD
```

Comment:
Does something need to change here?

Reply:
Yeah, this is not implemented yet. I need to do this in a separate PR with the caching.